handler.go

```diff
@@ -55,7 +68,67 @@ func (h handler) Create(ctx context.Context, request *pb.CreateUserRequest) (*pb
 	}, nil
 }
 
+func (h handler) Update(ctx context.Context, request *pb.UpdateUserRequest) (*pb.UpdateUserResponse, error) {
+	claims := request.GetClaims()
+	if claims == nil {
+		return nil, fmt.Errorf("update a user:\n%w", ErrNoPermission)
+	}
+
+	// Only admins or the user itself can update.
+	if !claims.IsAdmin && claims.UserId != request.Id {
+		return nil, fmt.Errorf("update a user:\n%w", ErrNoPermission)
+	}
+
+	// Only admins can change the IsAdmin flag.
+	if request.IsAdmin != nil && !claims.IsAdmin {
+		return nil, fmt.Errorf("update a user:\n%w", ErrNoPermission)
+	}
+
+	var passwordHash *string
+	if request.Password != nil {
+		var err error
+		newHash := ""
+		newHash, err = hash(request.Password.Value)
+		if err != nil {
+			return nil, fmt.Errorf("could not hash the password %w", err)
+		}
+
+		passwordHash = &newHash
+	}
+
+	var isAdmin *bool
+	if request.IsAdmin != nil {
+		isAdmin = &request.IsAdmin.Value
+	}
+
+	var username *string
+	if request.Username != nil {
+		username = &request.Username.Value
+	}
+
+	var email *string
+	if request.Email != nil {
+		email = &request.Email.Value
+	}
```

Comment:
Just a note: As soon as there are more fields to check, we may provide a helper function for less boilerplate code here:

```go
func stringOrNil(field *wrappers.StringValue) *string {
    if field != nil {
        return &field.Value
    }
    return nil
}

var username *string = stringOrNil(request.Username)
var email *string = stringOrNil(request.Email)
```

Reply:
As long as they are only a few it's ok I think.